src/store.ts

```diff
@@ -0,0 +1,22 @@
+import LruCache from "lru-cache";
+import type { Org } from "@app/base/orgs/Org";
+import type { EnsProfile } from "@app/base/registrations/registrar";
+import type { AddressType } from "@app/utils";
+
+export const ens = new LruCache<string, EnsProfile | null>({ max: 200 });
+export const type = new LruCache<string, AddressType>({ max: 200 });
+export const org = new LruCache<string, Org | null>({ max: 100 });
+export const request = new LruCache<string, any>({ max: 1000 });
+
+export async function cache<T>(key: string, store: LruCache<string, T>, query: (key: string) => Promise<T>): Promise<T> {
+  if (store.has(key)) {
+    const value = store.get(key);
+    if (value !== undefined) {
+      return value;
+    }
+  }
```

Comment:
I think the outer `if` is superfluous, if the key isn't in the cache, the stats for it won't be incremented and if it is, we have to get it anyway.
```suggestion
  const value = store.get(key);
  if (value !== undefined) {
    return value;
  }
```

Reply:
Technically I agree, the reason I added the `has` check is:

1. The return value of the `get` method and 2. the eventual update on the recency of the cache entry
> _From the lru-cache documentation_
> Will update the recency of the cache entry found.
>
> If the key is not found, get() will return undefined. This can be confusing when setting values specifically to undefined, as in cache.set(key, undefined). Use cache.has() to determine whether a key is present in the cache at all.

By using `has` as a first `if` I was able to get a boolean to check if the entry really existed, no matter if the entry would be `null`, `undefined` or another value, before trying to get it.

Reply:
Hm, in that case I'm confused by the inner `if (value !== undefined)`, because that means that we'll always hit the query branch in case the actual return value of the query is undefined instead of caching it.

Reply:
i c, yeah probably we should remove the `has` check and don't allow `undefined` cache entries.

Reply:
Done in 50b15b4fd0bd5a30461342f726bf4b24810118f9